File: backend/models/cowrie.py
from pydantic import BaseModel, ConfigDict


class CowrieSession(BaseModel):
    model_config = ConfigDict(extra='ignore')

    session_id: str
    timestamp: str
    src_ip: str
    src_port: int
    dst_port: int
    protocol: str
    username: str | None
    password: str | None
    commands: list[str]
    duration_seconds: int
    files_downloaded: list[str]
    country: str | None


class CowrieSample(BaseModel):
    model_config = ConfigDict(extra='ignore')

    sha256: str
    filename: str
    size: int
    downloaded_at: str
    src_ip: str
    url: str | None
    mime_type: str | None
    analyzed: bool
    analysis_id: str | None

File: packages/shared/types/cowrie.ts
export interface CowrieSession {
  session_id: string
  timestamp: string
  src_ip: string
  src_port: number
  dst_port: number
  protocol: string
  username: string | null
  password: string | null
  commands: string[]
  duration_seconds: number
  files_downloaded: string[]
  country: string | null
}

export interface CowrieSample {
  sha256: string
  filename: string
  size: number
  downloaded_at: string
  src_ip: string
  url: string | null
  mime_type: string | null
  analyzed: boolean
  analysis_id: string | null
}
